metrics-exporter/src/bin/io_engine/collector/nexus_stat.rs

```diff
@@ -0,0 +1,185 @@
+use super::init_volume_gauge_vec;
+use crate::{cache::Cache, client::nexus_stat::NexusIoStat, get_node_name};
+use prometheus::{
+    core::{Collector, Desc},
+    GaugeVec,
+};
+use std::{fmt::Debug, ops::Deref};
+use tracing::error;
+
+/// Collects Nexus IoStat metrics from cache.
+#[derive(Clone, Debug)]
+pub(crate) struct NexusIoStatsCollector {
+    nexus_bytes_read: GaugeVec,
+    nexus_num_read_ops: GaugeVec,
+    nexus_bytes_written: GaugeVec,
+    nexus_num_write_ops: GaugeVec,
+    nexus_read_latency_us: GaugeVec,
+    nexus_write_latency_us: GaugeVec,
+    descs: Vec<Desc>,
+}
+
+impl Default for NexusIoStatsCollector {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl NexusIoStatsCollector {
+    /// Initialize all the metrics to be defined for nexus iostat collector.
+    pub fn new() -> Self {
+        let mut descs = Vec::new();
+
+        let nexus_bytes_read =
+            init_volume_gauge_vec("bytes_read", "Total bytes read from the volume", &mut descs);
+        let nexus_num_read_ops = init_volume_gauge_vec(
+            "num_read_ops",
+            "Number of read operations on the volume",
+            &mut descs,
+        );
+        let nexus_bytes_written = init_volume_gauge_vec(
+            "bytes_written",
+            "Total bytes written on the volume",
+            &mut descs,
+        );
+        let nexus_num_write_ops = init_volume_gauge_vec(
+            "num_write_ops",
+            "Number of write operations on the volume",
+            &mut descs,
+        );
+        let nexus_read_latency_us = init_volume_gauge_vec(
+            "read_latency_us",
+            "Total read latency on the volume in usec",
+            &mut descs,
+        );
+        let nexus_write_latency_us = init_volume_gauge_vec(
+            "write_latency_us",
+            "Total write latency on the volume in usec",
+            &mut descs,
+        );
+
+        Self {
+            nexus_bytes_read,
+            nexus_num_read_ops,
+            nexus_bytes_written,
+            nexus_num_write_ops,
+            nexus_read_latency_us,
+            nexus_write_latency_us,
+            descs,
+        }
+    }
+}
+
+impl Collector for NexusIoStatsCollector {
+    fn desc(&self) -> Vec<&Desc> {
+        self.descs.iter().collect()
+    }
+
+    fn collect(&self) -> Vec<prometheus::proto::MetricFamily> {
+        let c = match Cache::get_cache().lock() {
```

Comment:
let use more descriptive variable names throughout the file rather than 1,2 letter like c, dp or x.

Reply:
Sure